mlrun/feature_store/api.py

```diff
@@ -410,6 +410,18 @@ def ingest(
         raise mlrun.errors.MLRunInvalidArgumentError(
             "feature set and source must be specified"
         )
+    if not targets:
+        if not isinstance(featureset, FeatureSet):
+            raise mlrun.errors.MLRunInvalidArgumentError(
```

Comment:
i don't quite like this two separate checks for feature set uri vs FeatureSet.
but looking at set_targets() i think that if the targets parameter to ingest() and featureset.spec.targets are both empty/None we can raise (ie we don't need featureset.spec.with_default_targets). Do you agree?

Reply:
The question is what we want to do when the user didn't use set_targets function at all... I think in this case we need to use the default targets, and this is the reason I am using featureset.spec.with_default_targets.

Reply:
Also if we get `featureset` from the use as uri (str) in this point it all-ready will be of  FeatureSet type.